blog: GSOC 2024 announcement post (#521)
electron/website: blog/2024-summer-of-code.md

---
title: Google Summer of Code 2024

> date: 2024-11-23T00:00:00.000Z

Reviewer: ```suggestion
date: 2024-02-23T00:00:00.000Z
```

Outside of Electron core, we also work on a variety of projects to help sustain the
Electron organization, such as:

- End-user distribution tooling (e.g. [Electron Forge](https://www.electronforge.io/)
  and [update.electronjs.org](https://github.com/electron/update.electronjs.org)).
- Learning materials for Electron developers (e.g. [electronjs.org](http://electronjs.org)
  and [Electron Fiddle](https://github.com/electron/fiddle)).
- Internal tools to streamline developer productivity (e.g. [Electron Build Tools](https://github.com/electron/build-tools)
  and [Sheriff](https://github.com/electron/sheriff)).

As a Summer of Code contributor, you would be collaborating with some of Electron’s core contributors
on one of many projects under the [github.com/electron](http://github.com/electron) umbrella.

## Before applying

If you aren’t very familiar with Electron, we would recommend you start by reading the
[documentation](https://electronjs.org/docs/latest) and trying out examples in [Electron Fiddle](https://electronjs.org/fiddle).

To learn more about Electron app distribution, you can also play around with

> [Electron Forge](https://github.com/electron/forge/blob/main/LICENSE) by creating a sample application:

Reviewer 2: ```suggestion
[Electron Forge](https://github.com/electron/forge/blob/main/README.md) by creating a sample application:
```

[A detailed guide of what to submit as part of your Electron application is here.](https://electronhq.notion.site/Electron-GSoC-2024-Contributor-Guidance-f1f4de7a0d9a4664a96c8d4dd70cb208?pvs=4)
Submit proposals directly to the Google Summer of Code portal. Note that proposals emailed to the
Electron team rather than submitted through the application portal will not be considered as a final submission.

If you want more guidance on your proposal or are unsure of what to include, we also recommend that
you follow [the official Google Summer of Code proposal writing advice here](https://google.github.io/gsocguides/student/writing-a-proposal).

Applications open on **March 18th, 2024** and close on **April 2nd, 2024**.

:::info

Our 2022 Google Summer of Code intern, [@aryanshridhar](https://github.com/aryanshridhar),
did an amazing job! If you want to see what Aryan worked on during his summer with Electron,
you can read his report in the [2022 GSoC program archives](https://summerofcode.withgoogle.com/archive/2022/organizations/electron).

:::

## Questions?

If you have questions we didn’t address in the blog post or inquiries for your proposal draft,

> please send us an email at [summer-of-code@electronjs.org](mailto:summer-of-code@electronjs.org)!

Reviewer 2: ```suggestion
please send us an email at [summer-of-code@electronjs.org](mailto:summer-of-code@electronjs.org) or check [GSoC FAQ](https://developers.google.com/open-source/gsoc/faq)!
```
